play.pokemonshowdown.com/src/panel-popups.tsx

```diff
@@ -1187,6 +1251,229 @@ class RegisterPanel extends PSRoomPanel {
+					Shaymin
+				</button>
+				<button onClick={this.setBg} value="solidblue" class="option">
+					<span class="bg" style="background: #344b6c"></span>Solid blue
+				</button>
+			</div>
+			<div style="clear: left"></div>
+			<p><strong>Custom</strong></p>
+			<p>
+				Drag and drop an image to PS (the background settings don't need to be
+				open), or upload:
+			</p>
+			<p><input type="file" accept="image/*" name="bgfile" /></p>
+			<p class="bgstatus"></p>
+			<p>
+				<button onClick={this.uploadBg} class="button"><strong>Done</strong></button>
+			</p>
+		</div>
+		</PSPanelWrapper>;
+	}
+}
+
+class ChatFormattingPanel extends PSRoomPanel {
+	static readonly id = 'chatformatting';
+	static readonly routes = ['chatformatting'];
+	static readonly location = 'semimodal-popup';
+	static readonly noURL = true;
+
+	handleOnChange = (ev: Event) => {
+		const setting = "hide" + (ev.currentTarget as HTMLInputElement).name;
+		const value = (ev.currentTarget as HTMLInputElement).checked;
+		let curPref = PS.prefs.chatformatting;
+		curPref[setting] = value;
+		PS.prefs.set("chatformatting", curPref);
+		ev.preventDefault();
+		ev.stopImmediatePropagation();
+	};
+
+	override render() {
+		const room = this.props.room;
+		return <PSPanelWrapper room={room} width={480}><div class="pad">
+			<p>Usable formatting:</p>
+			<p>**<strong>bold</strong>** (<kbd>Ctrl</kbd> + <kbd>B</kbd>)</p>
+			<p>__<em>italics</em>__ (<kbd>Ctrl</kbd> + <kbd>I</kbd>)</p>
+			<p>``<code>code formatting</code>`` (<kbd>Ctrl</kbd> + <kbd>`</kbd>)</p>
+			<p>~~<s>strikethrough</s>~~</p>
+			<p>^^<sup>superscript</sup>^^</p>
+			<p>\\<sub>subscript</sub>\\</p>
+			<p>
+				<label class="checkbox">
+					<input
+						onChange={this.handleOnChange}
+						type="checkbox"
+						name="greentext"
+						checked={PS.prefs.chatformatting.hidegreentext}
+					/> Suppress
+					<span class="greentext">&gt;greentext</span>
+				</label>
+			</p>
+			<p>
+				<label class="checkbox">
+					<input
+						onChange={this.handleOnChange}
+						type="checkbox"
+						name="me"
+						checked={PS.prefs.chatformatting.hideme}
+
+					/> Suppress <code>/me</code>
+					<em>action formatting</em></label>
+			</p>
+			<p>
+				<label class="checkbox">
+					<input
+						onChange={this.handleOnChange}
+						type="checkbox"
+						name="spoiler"
+						checked={PS.prefs.chatformatting.hidespoiler}
+					/> Auto-show spoilers:
+					<span class="spoiler">these things</span>
+				</label>
+			</p>
+			<p>
+				<label class="checkbox">
+					<input
+						onChange={this.handleOnChange}
+						type="checkbox"
+						name="links"
+						checked={PS.prefs.chatformatting.hidelinks}
+					/> Make [[clickable links]]
+					unclickable
+				</label>
+			</p>
+			<p>
+				<label class="checkbox">
+					<input
+						onChange={this.handleOnChange}
+						type="checkbox"
+						name="interstice"
+						checked={PS.prefs.chatformatting.hideinterstice}
+					/> Don't warn for
+					untrusted links
```

Comment:
```suggestion
					/> Don't warn for untrusted links
```

o_o how wide is your text editor? It should be at least 120cols.